### PR TITLE
Iminuit output

### PR DESCRIPTION
**Pull request by bkhelifi**
**Description**
This pull request is associated to the Issue #4383 permitting to fix the iminuit message.

### CODE REVIEW

**Comment by cbadams2** [comment]
I don't know if this is a functional solution to the issue. The way I'm reading the new code, I believe that it will still yield the same output for the problem case of choosing a fit strategy of 0:

It will check for `m.accurate`, find it to be `False`, and enter the first conditional statement, causing it to skip over `elif m.valid`. 

The trouble I think comes from associating the variable `success` with `minuit.accurate`, which is somewhat misleading because it seems that `accurate` is related to the covariance rather than convergence of the fit.  The iminuit code associates success with `minuit.valid` instead. I would maybe suggest the following, which is more or less the same as in https://github.com/scikit-hep/iminuit/blob/v2.21.0/src/iminuit/minimize.py#L124-L136:
```
def _get_message(m, parameters): 
    success = m.valid
    success &= np.all(np.isfinite([par.value for par in parameters])) 
    if success:
        message = "Optimization terminated successfully."
        if m.accurate:
            message += "."
        else:
            message += ", but uncertainties are unrealiable."
    else: 
        message = "Optimization failed." 
        fmin = m.fmin 
        if fmin.has_reached_call_limit: 
            message += " Call limit was reached." 
        if fmin.is_above_max_edm: 
            message += " Estimated distance to minimum too large." 
    return message 
```

**Review by AtreyeeS — APPROVED** [review]
Thanks @bkhelifi 
It would be nice to have this tested to ensure all the cases are correctly covered, but it might be complex coming up with a simple case where the fit fails

File: gammapy/modeling/iminuit.py [hunk 1]
```diff
@@ -175,12 +175,18 @@ def contour_iminuit(parameters, function, x, y, numpoints, sigma, **kwargs):
     }
 
 
-# this code is copied from https://github.com/iminuit/iminuit/blob/master/iminuit/_minimize.py#L95
+# This code is a mix from https://github.com/iminuit/iminuit/blob/master/iminuit/_minimize.py#L95
+# and https://github.com/scikit-hep/iminuit/blob/v2.21.0/src/iminuit/minimize.py#L124-L136
```
Comment (QRemy) [inline gammapy/modeling/iminuit.py:179]:
```suggestion
# This code is a copied from https://github.com/scikit-hep/iminuit/blob/v2.21.0/src/iminuit/minimize.py#L124-L136
```

The first link given as reference is invalid

**Review by registerrier — APPROVED** [review]
Thanks @bkhelifi . Looks good!

**Pull request merged**